### PR TITLE
Fix on-demand snapshots timing out; not honoring folder

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

* Fix on-demand snapshots not honoring folder
  Also fix etcd s3 tests to actually check that the files are saved to s3 🙃
* Fix 10 second etcd-snapshot request timeout
  The default clientaccess request timeout is too short. Wait longer by default, and add the s3 timeout if s3 is enabled.

#### Types of Changes ####

bugfix

#### Verification ####

See linked issue

#### Testing ####

yes

#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/9982
* https://github.com/k3s-io/k3s/issues/9985

#### User-Facing Change ####

```release-note
```

### CODE REVIEW

File: pkg/etcd/snapshot_handler.go [hunk 1]
```diff
@@ -170,14 +170,15 @@ func (e *ETCD) withRequest(sr *SnapshotRequest) *ETCD {
...
+		re.config.EtcdS3BucketName = sr.S3.Bucket
 		re.config.EtcdS3Endpoint = sr.S3.Endpoint
 		re.config.EtcdS3EndpointCA = sr.S3.EndpointCA
-		re.config.EtcdS3SkipSSLVerify = sr.S3.SkipSSLVerify
+		re.config.EtcdS3Folder = sr.S3.Folder
 		re.config.EtcdS3Insecure = sr.S3.Insecure
 		re.config.EtcdS3Region = sr.S3.Region
+		re.config.EtcdS3SecretKey = sr.S3.SecretKey
+		re.config.EtcdS3SkipSSLVerify = sr.S3.SkipSSLVerify
 		re.config.EtcdS3Timeout = sr.S3.Timeout.Duration
```
Comment (brandond) [inline pkg/etcd/snapshot_handler.go:182]: also sorted this to match with https://github.com/k3s-io/k3s/blob/d973fadbed133cf9c5db64c87ae724b1a80c766f/pkg/cli/etcdsnapshot/etcd_snapshot.go#L50-L60

**Review by fmoral2 — DISMISSED** [review]

**Comment by fmoral2** [comment]
time="2024-04-19T17:47:50Z" level=fatal msg="see server log for details: bucket test-bucket does not exist"

**Comment by brandond** [comment]
> time="2024-04-19T17:47:50Z" level=fatal msg="see server log for details: bucket test-bucket does not exist"

does the s3 mock only have a single bucket named `test`!?

**Review by aganesh-suse — APPROVED** [review]
LGTM.

**Review by matttrach — APPROVED** [review]

**Pull request merged**